File: programmers/다음에 올 숫자.py
""" 테스트 케이스
common = [1, 2, 3, 4]
common = [2, 4, 8]
common = [-16, -8, -4]
common = [-4, -6, -8]
common = [0,-2,-4, -6, -8]

"""
common = [-16, -8, -4, -2, -1]


def solution(common):
    if common[0] - common[1] == common[1] - common[2]:
        answer = common[len(common) - 1] + common[1] - common[0]
    else:
        answer = common[len(common) - 1] * common[2] // common[1]
    return answer


print(solution(common))

File: programmers/코딩테스트 입문 with Java/다음에올숫자.java
package 코딩테스트입문;

/**
 * https://school.programmers.co.kr/learn/courses/30/lessons/120924/solution_groups?language=java
 */
public class 다음에올숫자 {
    public static void main(String[] args) {
        System.out.println(solution(new int[]{1, 2, 3, 4}));
        System.out.println(solution(new int[]{2, 4, 8}));
        System.out.println(solution(new int[]{4, 16, 64}));
        System.out.println(solution(new int[]{4, 3, 2}));
        System.out.println(solution(new int[]{-2, -1, 0}));
    }

    public static int solution(int[] common) {
        int answer = 0;
        if (common[2] - common[1] == common[1] - common[0]) {
            answer = common[common.length - 1] + common[2] - common[1];
        } else {
            answer = common[common.length - 1] * common[2] / common[1];
        }
        return answer;
    }
}
